packages/dbml-connector/src/connectors/snowflakeConnector.ts

```diff
@@ -0,0 +1,370 @@
+
+    return acc;
+  }, {});
+
+  return {
+    tables: Object.values(tables),
+    fields,
+  };
+};
+
+const createConstraintKeysMap = (keys: Record<string, string>[], schemas: string[], constraintType: 'primary' | 'unique'): Record<string, constraintRow> => {
+  return keys.reduce((acc: Record<string, constraintRow>, row: Record<string, string>) => {
+    const { schema_name, table_name, column_name, constraint_name } = row;
+    const selectedSchema = schemas.length > 0 ? schemas.includes(schema_name) : true;
+    if (!selectedSchema) { return acc; }
+
+    const key = `${schema_name}.${table_name}.${constraint_name}`;
+
+    if (acc[key]) {
+      const columnNames = acc[key].columnNames;
+      columnNames.push(column_name);
+      return acc;
+    }
+
+    acc[key] = {
+      schemaName: schema_name,
+      tableName: table_name,
+      constraintName: constraint_name,
+      columnNames: [column_name],
+      type: '',
+    };
+    acc[key][constraintType] = true;
+
+    return acc;
+  }, {});
+};
+const generateIndexes = async (conn: Connection, databaseName: string, schemas: string[]) => {
+
+
+  const getPrimaryKeysSql = `
+    SHOW PRIMARY KEYS IN DATABASE ${databaseName};
+  `;
+
+  const getUniqueKeysSql = `
+    SHOW UNIQUE KEYS IN DATABASE ${databaseName};
+  `;
+
+  const primaryKeys = await executeQuery(conn, getPrimaryKeysSql);
+  const uniqueKeys = await executeQuery(conn, getUniqueKeysSql);
+
+  const primaryKeysByConstraint: Record<string, constraintRow> = createConstraintKeysMap(primaryKeys, schemas, 'primary');
+  const uniqueKeysByConstraint: Record<string, constraintRow> = createConstraintKeysMap(uniqueKeys, schemas, 'unique');
+
+  const allConstraints: constraintRow[] = [Object.values(primaryKeysByConstraint), Object.values(uniqueKeysByConstraint)].flat();
+  const { indexes, tableConstraints } = allConstraints.reduce((acc: generatedIndexes, row: constraintRow): generatedIndexes => {
+    const { schemaName, tableName, constraintName, columnNames, type, primary, unique } = row;
+    const key = `${schemaName}.${tableName}`;
+
+    if (columnNames.length < 2) {
+      acc.tableConstraints[key] = {
+        [columnNames[0]]: {
+          pk: primary,
+          unique,
+        },
+      };
+      return acc;
+    }
+
+    const index: Index = {
+      name: constraintName,
+      type,
+      unique,
+      pk: primary,
+      columns: columnNames.map((columnName) => ({
+        type: 'column',
+        value: columnName,
+      })),
+    };
+
+    if (acc.indexes[key]) {
+      acc.indexes[key].push(index);
+    } else {
+      acc.indexes[key] = [index];
+    }
+
+    return acc;
+  }, { indexes: {}, tableConstraints: {}});
+
+  return {
+    indexes,
+    tableConstraints,
+  };
+};
+
+const fetchSchemaJson = async (connection: string): Promise<DatabaseSchema> => {
+  const config = parseConnectionString(connection);
+  const conn = await connectToSnowflake(config);
+  if (conn instanceof Error) {
```

Comment:
Does snowflake returns error instead of throwing it?
If it throws the error, you do not need to check here